dockerfiles/Dockerfile.deb

```diff
@@ -28,4 +28,16 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update \
         m4 \
         file \
         ssh \
-        valgrind
+        valgrind \
+        wget
+
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.15.7/cmake-3.15.7.tar.gz \
+    && tar -xzvf cmake-3.15.7.tar.gz \ 
+    && cd cmake-3.15.7 \
+    && ./bootstrap \
+    && make -j 2 \
+    && make install
```

Comment:
Same here; WDYT about allowing the version to be specified, switching to `curl` (or ensure no certificate checks), and cleanup the files?

```suggestion
ARG CMAKE_VERSION=3.15.7
RUN curl -sSL https://github.com/Kitware/CMake/releases/download/v$CMAKE_VERSION/cmake-$CMAKE_VERSION.tar.gz  -o /tmp/cmake-$CMAKE_VERSION.tar.gz \
    && tar -xzvf /tmp/cmake-$CMAKE_VERSION.tar.gz -C /tmp \ 
    && pushd /tmp/cmake-$CMAKE_VERSION \
    && ./bootstrap \
    && make -j 2 \
    && make install \
    && popd \
    && rm -rf /tmp/cmake-$CMAKE_VERSION.tar.gz /tmp/cmake-$CMAKE_VERSION
```

or

```suggestion
ARG CMAKE_VERSION=3.15.7
RUN wget --no-check-certificate -O /tmp/cmake-$CMAKE_VERSION.tar.gz https://github.com/Kitware/CMake/releases/download/v$CMAKE_VERSION/cmake-$CMAKE_VERSION.tar.gz \
    && tar -xzvf /tmp/cmake-$CMAKE_VERSION.tar.gz -C /tmp \ 
    && pushd /tmp/cmake-$CMAKE_VERSION \
    && ./bootstrap \
    && make -j 2 \
    && make install \
    && popd \
    && rm -rf /tmp/cmake-$CMAKE_VERSION.tar.gz /tmp/cmake-$CMAKE_VERSION
```


You might also consider changing the `make -j 2` to use the actual number of cores available or remove the `-j 2` all together; unless of course we are guaranteed that 2 cores are always available:

#### without `nproc`

This is very ugly and unpredictable, but gives you the actual number of physical cores

```
grep ^cpu\\scores /proc/cpuinfo | uniq |  awk '{print $4}'
```

#### with `nproc`

This is a lot simpler and won't break if the `/proc/cpuinfo` format changes. The only negative here is that it will return all cores if hyperthreading is enabled.

**Note**: Ensure `coreutils` is installed

```
nproc
```